src/html-grammar.pegjs

```diff
@@ -548,6 +549,45 @@ text_node "Text Node"
   };
 }
 
+/* Inline php code */
+php "PHP Code"
+  = php_open com:(php_block) cc:(php_close)?
+  {
+    if (cc === null) {
+      return error('Found an open PHP code tag without a closing tag');
+    }
+    return com;
+  }
+
+php_open "PHP start"
+  = "<?php"
+  
+php_close "PHP close"
+  = "?>"
+  / EOF
+  
+php_block
+  = s pb:php_scan s
+  {
+    var tn = pb !== null ? _u.textNode(pb) : '';
+    if(tn.indexOf('?>') !== -1) {
+      return error("Cannot have a php closing block " + esc.val('?>') + " inside of a PHP code block");
+    }
+    return {
```

Comment:
Could we change the output format of the PHP tag so it ends up returning this format:

``` javascript
{
  "type": "php",
  "attributes": {},
  "contents":"var x = 23;"
}
```